services/web/src/helpers/modal.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { Modal } from 'semantic';
+
+export default function modal(Component) {
+  class ModalWrapper extends React.Component {
+    state = {
+      open: this.props.open,
+      modelKey: Date.now(),
+    };
+
+    componentDidUpdate(lastProps) {
+      const { open } = this.props;
+      if (open && open !== lastProps.open) {
```

Comment:
I realize you're just moving this around but could we change this to `'open' in props` instead? The props should also be able to close the modal.

Reply:
not sure i follow e.g. wouldnt open: false close the modal?

Reply:
Yeah, it's probably unlikely but having the props override the state, not just to open... actually now that I think about it `'open' in props` isn't needed just `if (open !== lastProps.open)` is enough I think

Reply:
yeah sorry i wasn't clear, i came to the same conclusion i think, see the latest code